File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/AXNPC.java
package Auxiliary_Modules;

public class AXNPC {
    public RefreshQ responder = new RefreshQ();
    public PercentDripper dripper = new PercentDripper();
    public AXCmdBreaker cmdBreaker = new AXCmdBreaker("say");

    public AXNPC(int replyStockLim, int outputChance) {
        responder.setLimit(replyStockLim);
        if (outputChance >0 && outputChance < 101){
            dripper.setLimis(outputChance);
        }
    }

    public String respond(){
        if (dripper.drip()){
            return responder.getRNDElement();
        }
        return "";
    }
    public String respondPlus(int plus){
        // increase rate of output
        if (dripper.dripPlus(plus)){
            return responder.getRNDElement();
        }
        return "";
    }
    public boolean learn(String ear){
        // say hello there : hello there is learned
        String temp = cmdBreaker.extractCmdParam(ear);
        if (temp.isEmpty()){return false;}
        responder.add(temp);
        return true;
    }
    public String strRespond(String ear){
        // respond if ear contains a learned input
        if (ear.isEmpty()){return "";}
        if (dripper.drip() && responder.strContainsResponse(ear)){
            return responder.getRNDElement();
        }
        return "";
    }
    public String forceRespond(){
        return responder.getRNDElement();
    }
    public void setConjuration(String conjuration){
        cmdBreaker.conjuration = conjuration;
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/AXNPC.kt
package auxiliary_modules

open class AXNPC(replyStockLim: Int, outputChance: Int) {
    var responder = RefreshQ()
    var dripper = PercentDripper()
    var cmdBreaker = AXCmdBreaker("say")

    init {
        responder.limit = replyStockLim
        if (outputChance in 1..100) {
            dripper.setLimis(outputChance)
        }
    }

    fun respond(): String {
        return if (dripper.drip()) {
            responder.rNDElement
        } else ""
    }

    fun respondPlus(plus: Int): String {
        // increase rate of output
        return if (dripper.dripPlus(plus)) {
            responder.rNDElement
        } else ""
    }

    fun learn(ear: String?): Boolean {
        // say hello there : hello there is learned
        val temp = cmdBreaker.extractCmdParam(ear!!)
        if (temp.isEmpty()) {
            return false
        }
        responder.add(temp)
        return true
    }

    fun strRespond(ear: String): String {
        // respond if ear contains a learned input
        if (ear.isEmpty()) {
            return ""
        }
        return if (dripper.drip() && responder.strContainsResponse(ear)) {
            responder.rNDElement
        } else ""
    }

    fun forceRespond(): String {
        return responder.rNDElement
    }

    fun setConjuration(conjuration: String) {
        cmdBreaker.conjuration = conjuration
    }
}
